banderwagon/element.go

```diff
@@ -116,49 +116,41 @@ func (p *Element) SetBytesTrusted(buf []byte) error {
 
 // computes X/Y
 func (p Element) mapToBaseField() fp.Element {
-
 	var res fp.Element
 	res.Div(&p.inner.X, &p.inner.Y)
 	return res
 }
 
-func (p Element) MapToScalarField() fr.Element {
+func (p Element) MapToScalarField(res *fr.Element) {
 	basefield := p.mapToBaseField()
 	baseFieldBytes := basefield.BytesLE()
 
-	var res fr.Element
 	res.SetBytesLE(baseFieldBytes[:])
-
-	return res
 }
```

Comment:
@kevaundray, I also changed your original API for the single-value case. Previous L129 (red) escapes to the heap.